bottlecap/src/logs/lambda/processor.rs

```diff
@@ -220,67 +220,91 @@ impl LambdaProcessor {
     }
 
     fn get_intake_log(&mut self, mut lambda_message: Message) -> Result<IntakeLog, Box<dyn Error>> {
-        let request_id = match lambda_message.lambda.request_id {
-            // Log already has a `request_id`
+        // Assign request_id from message or context if available
+        lambda_message.lambda.request_id = match lambda_message.lambda.request_id {
             Some(request_id) => Some(request_id.clone()),
-            // Default to the `request_id` we've seen so far, if any.
-            None => (!&self.invocation_context.request_id.is_empty())
-                .then(|| self.invocation_context.request_id.clone()),
+            None => {
+                if self.invocation_context.request_id.is_empty() {
+                    None
+                } else {
+                    Some(self.invocation_context.request_id.clone())
+                }
+            }
```

Comment:
Is this change necessary?

Reply:
Clippy:
```
warning: unnecessary boolean `not` operation
   --> src/logs/lambda/processor.rs:227:17
    |
227 | /                 if !self.invocation_context.request_id.is_empty() {
228 | |                     Some(self.invocation_context.request_id.clone())
229 | |                 } else {
230 | |                     None
231 | |                 }
    | |_________________^
    |
    = help: remove the `!` and swap the blocks of the `if`/`else`

```

Reply:
I mean, what about the previous code?

Reply:
Oh, no it's not strictly necessary, I just thought it was easier to understand.